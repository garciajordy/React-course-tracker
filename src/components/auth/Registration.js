import React, { useState } from 'react'
import axios from 'axios';
import { useHistory } from 'react-router-dom';
import store from "../../store/Store"
import logIn from "../../actions/LoginAction";

function Registration() {
    let history = useHistory()
    const [data, setData] = useState({
        email: '',
        password: '',
        password_confirmation: '',
        registrationErrors: ''
    })
    function handleSubmit(event) {
        axios.post("http://localhost:3001/registrations", {
            user: {
                email: data.email,
                password: data.password,
                password_confirmation: data.password_confirmation
            }
        },
        { withCredentials: true }
        ).then(response => {
            if (response.data.status === "created") {
                store.dispatch(logIn(response.data.user))
                history.push("/dashboard")
            };
        }).catch(error => {
            console.log("registration error", error)
            if (data.password.length < 6) {
                setData((prev => ({
                    ...prev,
                    registrationErrors: "Password needs a minimum of 6 characters."
                })))
            }
            else if (data.password !== data.password_confirmation) {
                console.log()
                setData((prev => ({
                    ...prev,
                    registrationErrors: "Password confirmation is not correct!."
                })))
            } else {

                setData((prev => ({
                    ...prev,
                    registrationErrors: "Email has already been taken."
                })))
            }
        })
        event.preventDefault();
    }

    function handleChange(e) {
        setData((prev => ({...prev, [e.target.name]: e.target.value})))
    }

    return (
        <div>
            <form onSubmit={handleSubmit} >
               <p> {data.registrationErrors}</p>
            <input type='email' name='email' placeholder='Email' value={data.email} onChange={handleChange} required />
            <input type='password' name='password' placeholder='Password' value={data.password} onChange={handleChange} required />
                <input type='password' name='password_confirmation' placeholder='Password_confirmation' value={data.password_confirmation} onChange={handleChange} required />
                <button type="submit">Register</button>
            </form>
        </div>
    )
}

export default Registration

