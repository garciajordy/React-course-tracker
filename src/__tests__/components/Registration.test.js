import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Registration from '../../components/auth/Registration';
import store from '../../store/Store';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<Provider store={store}><Registration /></Provider>);
  expect(asFragment(<Provider store={store}><Registration /></Provider>)).toMatchSnapshot();
});

it('displays heading', () => {
  render(<Provider store={store}><Registration /></Provider>);
  expect(screen.getByText('Register')).toBeTruthy();
});
