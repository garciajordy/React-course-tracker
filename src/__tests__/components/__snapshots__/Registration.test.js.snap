// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <p>
         
      </p>
      <input
        name="email"
        placeholder="Email"
        required=""
        type="email"
        value=""
      />
      <input
        name="password"
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <input
        name="password_confirmation"
        placeholder="Password_confirmation"
        required=""
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Register
      </button>
    </form>
  </div>
</DocumentFragment>
`;
